### PR TITLE
Create dictionary_from_trie_C++.md

### LINKED ISSUE #24: Dictionary using Trie
# Description

Implement a dictionary using a [trie](https://www.hackerearth.com/practice/data-structures/advanced-data-structures/trie-keyword-tree/tutorial/) in which the user enters a word, if it is present in the prebuilt dictionary print its meaning/definition. The user should also be able to include words of his/her own choice in the  #dictionar.

## Required prebuilt words
- hacktoberfest
- git
- open-source
- issue
- contribution

## Language
- [x] C++: @dwij2212 
- [x] Java
- [x] Python: @arnab031

## Type of issue

- [x] Feature
- [ ] Bug

The checklist will be updated by the maintainers/author of this issue according to the language in which snippets are added or are proposed to be added.

### PR DESCRIPTION
**Pull request by dwij2212**
# Title/Heading
Dictionary from trie
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes: #24 

## Type of change

Please delete options that are not relevant. (including this line)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generates no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by deepika-rana — COMMENTED** [review]
@dwij2212 The required prebuilt words are missing. Also, the user's choice word search is not implemented.

File: challenges/dictionary_from_trie_C++.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	char key = str[index];
+	// recursive definition to iterate through trie
+	return search(temp->branches[key], str, index+1);
+}
+
+```
+
+## Driver Code
```
Comment (kaustubhgupta) [inline challenges/dictionary_from_trie_C++.md]: The driver code heading doesn't require any explicit markdown level. Make it as a comment in the code section itself.

File: challenges/dictionary_from_trie_C++.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    insert(root, "book", "a written or printed work");  
+    insert(root, "draw", "to do a picture or diagram of something "); 
+  
+  	// search for "book" in our trie
+    string str = "book"; 
+    cout << search(root, str); 
+  
+    return 0; 
+}
+```
```
Comment (kaustubhgupta) [inline challenges/dictionary_from_trie_C++.md:101]: After the code section, we require contributor details. Check out other snippets for a better idea

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: challenges/dictionary_from_trie_C++.md (hunk 2 above)
Comment (dwij2212) [reply challenges/dictionary_from_trie_C++.md:101]: Yeah, I forgot that, I'll add it in the next commit.

**Comment by dwij2212** [comment]
@deepika-rana I have included some pre-built words but how am I supposed to get the contents of an entire dictionary? To add some words the user needs to just add the line `insert(root, word, meaning);`. To search the user needs to just add a line `cout<<search(root, "word_to_search");`

**Comment by deepika-rana** [comment]
Okay, to search for a word the string is declared in the code itself. You can rather custom input a string `cin>>word_to_search` if you wish.

**Review by deepika-rana — APPROVED** [review]

**Review by kaustubhgupta — APPROVED** [review]

**Pull request merged**